crates/hotshot/types/src/epoch_membership.rs

```diff
@@ -36,6 +37,9 @@ pub struct EpochMembershipCoordinator<TYPES: NodeType> {
     /// wait for the actual catchup and allert future callers when it's done
     catchup_map: Arc<Mutex<EpochMap<TYPES>>>,
 
+    /// Callback function to store a drb result when one is calculated during catchup
+    storage_add_drb_result_fn: Option<StorageAddDrbResultFn<TYPES>>,
+
```

Comment:
Is there a reason to not just take the storage, I feel like having a callback here, that pretty much just calls storage to add DRB doesn't gain us much and just makes the code a bit confusing.